src/cpp/session/modules/SessionLinter.cpp

```diff
@@ -99,41 +133,138 @@ void addInferredSymbols(std::set<std::string>* pSymbols)
 
 void addNamespaceSymbols(std::set<std::string>* pSymbols)
 {
+   // Add symbols specifically mentioned as 'importFrom'
+   // directives in the NAMESPACE.
    BOOST_FOREACH(const std::set<std::string>& symbolNames,
                  RSourceIndex::getImportFromDirectives() | boost::adaptors::map_values)
    {
       pSymbols->insert(symbolNames.begin(), symbolNames.end());
    }
+   
+   // Make all (exported) symbols published by packages
+   // that are 'import'ed in the NAMESPACE.
+   BOOST_FOREACH(const std::string& package,
+                 RSourceIndex::getImportedPackages())
+   {
+      const AsyncLibraryCompletions& completions =
+            RSourceIndex::getCompletions(package);
+      pSymbols->insert(
+               completions.exports.begin(),
+               completions.exports.end());
+   }
 }
 
-Error getAllAvailableRSymbols(const FilePath& filePath,
-                              std::set<std::string>* pSymbols)
+void fillNamespace(const std::string& nsName,
+                   std::vector<std::string>* pSymbols)
 {
-   using namespace r::exec;
+   r::sexp::Protect protect;
+   SEXP ns = r::sexp::findNamespace(nsName);
+   Error error = r::sexp::getNamespaceExports(ns, pSymbols);
+   if (error)
+      LOG_ERROR(error);
+}
+
+void addBaseSymbols(std::set<std::string>* pSymbols)
+{
+
+#define RS_GET_EXPORTED_SYMBOLS(__NAMESPACE__)                                 \
```

Comment:
If there is a way to do this with a function call that is *much* preferred even if it's slightly more typing